src/main/java/com/paneedah/mwc/groovyscript/script/CraftingStation.java

```diff
@@ -0,0 +1,177 @@
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import com.paneedah.mwc.groovyscript.MWCContainer;
+import com.paneedah.mwc.groovyscript.recipes.GSCrafting;
+import com.paneedah.weaponlib.crafting.CraftingEntry;
+import com.paneedah.weaponlib.crafting.CraftingGroup;
+import com.paneedah.weaponlib.crafting.CraftingRegistry;
+import com.paneedah.weaponlib.crafting.IModernCrafting;
+import net.minecraft.item.ItemStack;
+
+import java.util.ArrayList;
+
+import static com.paneedah.weaponlib.crafting.CraftingGroup.*;
+import static com.paneedah.weaponlib.crafting.CraftingRegistry.*;
+
+public class CraftingStation extends VirtualizedRegistry<IModernCrafting> {
+
+    @Override
+    public void onReload() {
+        this.removeScripted().forEach(CraftingRegistry::deleteRecipeRegistry);
+        this.restoreFromBackup().forEach(CraftingRegistry::registerRecipe);
+    }
+
+    /**
+     * Remove ALL recipes. no matter the category.
+     */
+
+    public void removeAll() {
+        for (ArrayList<IModernCrafting> list : craftingMap.values()) {
+            for (IModernCrafting recipe : list) {
+                deleteRecipeRegistry(recipe);
+                this.addBackup(recipe);
+            }
+        }
+    }
+
+    /**
+     * Remove Recipe that outputs given Ingredient.
+     *
+     * @param ingredient Target Ingredient. any recipe with matching output will be removed.
+     */
+
+    public void remove(IIngredient ingredient) {
+        for (ArrayList<IModernCrafting> list : craftingMap.values()) {
+            for (IModernCrafting recipe : list) {
+                if (ingredient.test(recipe.getItem())) {
+                    deleteRecipeRegistry(recipe);
+                    this.addBackup(recipe);
+                }
+            }
+        }
+    }
+
+    /**
+     * Remove ALL recipes of Given Category.
+     *
+     * @param group CraftingGroup to remove all recipes inside. valid values = "GUN", "ATTACHMENT_NORMAL", "ATTACHMENT_MODIFICATION", "BULLET", "MAGAZINE"
+     */
+    public void removeAllinGroup(String group) {
+        clearGroup(CraftingGroup.valueOf(group));
+    }
+
+    public void removeinGroup(IIngredient ingredient, CraftingGroup group) {
+        for (IModernCrafting recipe : craftingMap.get(group)) {
+            if (ingredient.test(recipe.getItem())) {
+                deleteRecipeRegistry(recipe);
+                this.addBackup(recipe);
+            }
+        }
+    }
+
+    public void removeinGroup(IIngredient ingredient, String group) {
+        removeinGroup(ingredient, CraftingGroup.valueOf(group));
+    }
+
+    public void addRecipe(IModernCrafting crafting) {
+        registerRecipe(crafting);
+        this.addScripted(crafting);
+    }
+
+    /**
+     * Start Recipe Builder for GroovyScript.
+     */
+
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
```

Comment:
Again why make a method that just calls a ctor and nothing else just call the ctor directly?

Reply:
the reason why I left ``buildRecipe()`` is it makes calling recipebuilder much easier.
so with this we can do this:
```groovy
mods.mwc.craftingstation.recipeBuilder().output(item('minecraft:clay')).addEntry(item('minecraft:nether_star')*4).register()
```
instead of this
```groovy
import com.paneedah.mwc.groovyscript.script.CraftingStation.RecipeBuilder
new RecipeBuilder().output(item('minecraft:clay')).addEntry(item('minecraft:nether_star')*4).register()
```

Reply:
Import is necessary either way, second is more consistent with the rest of the codebase and easier to type

Reply:
in first case Groovyscript handles the import statement automatically. so you don't need to type it manually.